packages/trace-viewer/src/ui/errorsTab.tsx

```diff
@@ -95,43 +97,101 @@ export function useErrorsTabModel(model: modelUtil.MultiTraceModel | undefined):
   }, [model]);
 }
 
+function Error({ message, error, errorId, sdkLanguage, pageSnapshot, revealInSource }: { message: string, error: ErrorDescription, errorId: string, sdkLanguage: Language, pageSnapshot?: string, revealInSource: (error: ErrorDescription) => void  }) {
+  const [showLLM, setShowLLM] = React.useState(false);
+  const llmAvailable = useIsLLMAvailable();
+  const gitCommitInfo = useGitCommitInfo();
+  const diff = gitCommitInfo?.['pull.diff'] ?? gitCommitInfo?.['revision.diff'];
+
+  let location: string | undefined;
+  let longLocation: string | undefined;
+  const stackFrame = error.stack?.[0];
+  if (stackFrame) {
+    const file = stackFrame.file.replace(/.*[/\\](.*)/, '$1');
+    location = file + ':' + stackFrame.line;
+    longLocation = stackFrame.file + ':' + stackFrame.line;
+  }
+
+  return <div style={{ display: 'flex', flexDirection: 'column', overflowX: 'clip' }}>
+    <div className='hbox' style={{
+      alignItems: 'center',
+      padding: '5px 10px',
+      minHeight: 36,
+      fontWeight: 'bold',
+      color: 'var(--vscode-errorForeground)',
+      flex: 0,
+    }}>
+      {error.action && renderAction(error.action, { sdkLanguage })}
+      {location && <div className='action-location'>
+        @ <span title={longLocation} onClick={() => revealInSource(error)}>{location}</span>
+      </div>}
+      <span style={{ position: 'absolute', right: '5px' }}>
+        {llmAvailable
+          ? <FixWithAIButton conversationId={errorId} onChange={setShowLLM} value={showLLM} error={message} diff={diff} pageSnapshot={pageSnapshot} />
+          : <CopyPromptButton error={message} pageSnapshot={pageSnapshot} diff={diff} />}
+      </span>
+    </div>
+
+    <ErrorMessage error={message} />
+
+    {showLLM && <AIConversation conversationId={errorId} />}
+  </div>;
+}
+
+function FixWithAIButton({ conversationId, value, onChange, error, diff, pageSnapshot }: { conversationId: string, value: boolean, onChange: React.Dispatch<React.SetStateAction<boolean>>, error: string, diff?: string, pageSnapshot?: string }) {
+  const chat = useLLMChat();
+
+  return <ToolbarButton
+    onClick={() => {
+      if (!chat.getConversation(conversationId)) {
+        const conversation = chat.startConversation(conversationId, [
+          `My Playwright test failed. What's going wrong?`,
```

Comment:
Is this different from `fixTestPrompt`? I feel like we should probably reuse them?

Reply:
It is different, because we can make a difference between system prompts, visible prompt contents and actual prompt contents. None of that works with the copy prompt buttons.